### PR TITLE
Add tests for game board builder; add method for stopping game loop

### PR DESCRIPTION
**Pull request by piotr921**
Add tests for game board builder. Add method for stopping game loop in class Game.

### CODE REVIEW

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 1]
```diff
@@ -102,19 +102,22 @@ public void onNameTextFieldEntered() {
...
-        if(scene == null) {return;}
+        Scene scene = mainBorderPane.getScene();
+        if (scene == null) {
+            return;
+        }
         Window window = scene.getWindow();
         window.setOnCloseRequest(event -> {
-            if(game != null) {
+            if (game != null) {
+                game.stopGameRunning();
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: 👍

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 2]
```diff
@@ -145,10 +148,12 @@ void onSendToServerButtonClicked() {
         game.buildOpponentsBoard(new OpponentBoardController(
                 new GameBoard(opponentsGridPane), game));
         game.sendTheFleetToServer();
-        new Thread( () -> {try {
-            game.runTheGame();
-        } catch (InterruptedException e) {
-            logger.log(DEFAULT_LEVEL, "Game interrupted");
-        }}).start();
+        new Thread(() -> {
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: 👍

File: src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java [hunk 3]
```diff
@@ -40,9 +40,8 @@ public GameBoardBuilder(OpponentBoardController boardController) {
         this.gridPane = boardController.getBoardGridPane();
     }
 
-    public GameBoard buildGameBoard() {
+    public void buildGameBoard() {
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java]: 👍

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 4]
```diff
@@ -145,10 +148,12 @@ void onSendToServerButtonClicked() {
...
+            try {
+                game.runTheGame();
+            } catch (InterruptedException e) {
+                logger.log(DEFAULT_LEVEL, "Game interrupted");
```
Comment (Ice94) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: :1st_place_medal:

File: src/test/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilderTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    public void setupJavaFx() {
+        final CountDownLatch latch = new CountDownLatch(1);
+        SwingUtilities.invokeLater(() -> {
+            new JFXPanel(); // initializes JavaFX environment
+            latch.countDown();
+        });
+    }
+
+    @Test
+    public void shouldBuildGameBoard() throws InterruptedException {
```
Comment (Ice94) [inline src/test/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilderTest.java]: I like this name.

File: src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java [hunk 6]
```diff
@@ -40,9 +40,8 @@ public GameBoardBuilder(OpponentBoardController boardController) {
...
         fillTheBoardWithButtonsAndLabels();
```
Comment (paulstarbi) [inline src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java]: you don't need both methods.

**Pull request merged**